client/asset/btc/btc.go

```diff
@@ -705,72 +705,149 @@ func (btc *ExchangeWallet) feeRateWithFallback(confTarget uint64) uint64 {
+		avail += utxo.amount
+	}
+
+	val := lots * lotSize
+
+	sum, inputsSize, _, _, _, _, err := btc.fund(val, lots, utxos, nfo)
+	if err != nil {
+		return nil, false, err
+	}
+
+	reqFunds := calc.RequiredOrderFundsAlt(val, uint64(inputsSize), lots, nfo.SwapSizeBase, nfo.SwapSize, nfo.MaxFeeRate)
+	maxFees := reqFunds - val
+
+	estHighFunds := calc.RequiredOrderFundsAlt(val, uint64(inputsSize), lots, nfo.SwapSizeBase, nfo.SwapSize, networkFeeRate)
+	estHighFees := estHighFunds - val
+
+	estLowFunds := calc.RequiredOrderFundsAlt(val, uint64(inputsSize), 1, nfo.SwapSizeBase, nfo.SwapSize, networkFeeRate)
+	if btc.segwit {
+		estLowFunds += dexbtc.P2WSHOutputSize * (lots - 1) * networkFeeRate
+	} else {
+		estLowFunds += dexbtc.P2SHOutputSize * (lots - 1) * networkFeeRate
+	}
+
+	estLowFees := estLowFunds - val
+
+	// Math for split transactions is a little different.
+	if trySplit {
+		_, extraMaxFees := btc.splitBaggageFees(nfo.MaxFeeRate)
+		_, splitFees := btc.splitBaggageFees(networkFeeRate)
+
+		if avail >= reqFunds+extraMaxFees {
+			return &asset.SwapEstimate{
+				Lots:               lots,
+				Value:              val,
+				MaxFees:            maxFees + extraMaxFees,
+				RealisticBestCase:  estLowFees + splitFees,
+				RealisticWorstCase: estHighFees + splitFees,
+				Locked:             val + maxFees + extraMaxFees,
+			}, true, nil
+		}
+	}
+
+	return &asset.SwapEstimate{
+		Lots:               lots,
+		Value:              val,
+		MaxFees:            maxFees,
+		RealisticBestCase:  estLowFees,
+		RealisticWorstCase: estHighFees,
+		Locked:             sum,
+	}, false, nil
+}
+
+// PreRedeem generates an estimate of the range of redemption fees that could
+// be assessed.
+func (btc *ExchangeWallet) PreRedeem(req *asset.PreRedeemForm) (*asset.PreRedeem, error) {
 	feeRate := btc.feeRateWithFallback(btc.redeemConfTarget)
-	var size uint64 = dexbtc.MinimumTxOverhead + dexbtc.TxInOverhead + dexbtc.TxOutOverhead
+	// Best is one transaction with req.Lots inputs and 1 output.
+	var best uint64 = dexbtc.MinimumTxOverhead
+	// Worst is req.Lots transactions, each with one input and one output.
+	var worst uint64 = dexbtc.MinimumTxOverhead * req.Lots
+	var inputSize, outputSize uint64
 	if btc.segwit {
 		// Add the marker and flag weight here.
-		var witnessVBytes uint64 = (dexbtc.RedeemSwapSigScriptSize + 2 + 3) / 4
-		size += witnessVBytes + dexbtc.P2WPKHOutputSize
+		inputSize = dexbtc.TxInOverhead + (dexbtc.RedeemSwapSigScriptSize+2+3)/4
+		outputSize = dexbtc.P2WPKHOutputSize
+
 	} else {
-		size += dexbtc.RedeemSwapSigScriptSize + dexbtc.P2PKHOutputSize
+		inputSize = dexbtc.TxInOverhead + dexbtc.RedeemSwapSigScriptSize
+		outputSize = dexbtc.P2PKHOutputSize
 	}
-	return size * feeRate, nil
+	best += inputSize*req.Lots + outputSize
+	worst += (inputSize + outputSize) * req.Lots
+
+	return &asset.PreRedeem{
+		Estimate: &asset.RedeemEstimate{
+			RealisticWorstCase: worst * feeRate,
+			RealisticBestCase:  best * feeRate,
```

Comment:
All we can really say before matching is that the fees will (realistically) fall in a range between the best and worst case. Best case provides the lower limit. Worst cast provides the upper limit.

Reply:
Just to expand on this a little, the worst case is probably the one I would pay attention to the most, but for users who place large orders, they may want to collect statistics over time to better estimate where their risk falls between these two points. 

From a UI perspective, this will allow us to offer some additional clarity on fees instead of just showing the worst-case value.